refactor: Use HeaderTitle component on RoomHeader

abhishek0450 opened the pull request:
<!-- This is a pull request template, you do not need to uncomment or remove the comments, they won't show up in the PR text. -->
This PR introduces a new reusable HeaderTitle component to create consistent and pressable headers across the application, it uses PlatformPressable for onPress handling. This refactor simplifies the RoomHeader and promotes code reuse. 

## Proposed changes
This pull request refactors the RoomHeader to use a new, reusable HeaderTitle component. The key motivation for this change is to make the room header title pressable and to improve code modularity.

The new HeaderTitle component encapsulates the logic for displaying the room title and handles press events via the new PlatformPressable utility, ensuring consistent behavior across platforms.

## Issue(s) 
Closes #6255

## How to test or reproduce
1. Go to any chat room.
2. The header of the room should be visible.
3. Tap on the header title.
4. The action to view Room Info should be triggered.
5. Verify that the header appears and functions correctly on both iOS and Android.

## Screenshots
<!-- Please add any relevant screenshots here. -->

https://github.com/user-attachments/assets/c3209cd3-6560-4218-9f98-5e02df22e7e3



## Types of changes
<!-- What types of changes does your code introduce to Rocket.Chat? -->
<!-- Put an x in the boxes that apply -->

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist
<!-- Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code. -->

- [x] I have read the CONTRIBUTING doc
- [x] I have signed the CLA
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments
The main goal was to create a more modular and reusable header component. By extracting the title into its own HeaderTitle component and using a PlatformPressable wrapper, we've made the UI more consistent and easier to maintain.

OtavioStasiak reviewed (changes requested):
We already have a Header Title here: `app/containers/Header/components/HeaderTitle`.
The goal of this issue is to implement `onPress` in the `HeaderTitle`.

abhishek0450 commented:
Hi , 
So I replaced RoomHeader's inline HeaderTitle implementation with the existing reusable HeaderTitle component and implement onPress to it, by creating a new PlatformPressable component to provide cross-platform touch feedback with platform-specific opacity values, replacing TouchableOpacity usage. I also fixed text styles in HeaderTitle components to prevent text flow out of its box .

OtavioStasiak commented on app/containers/RoomHeader/RoomHeader.tsx:
```diff
@@ -209,13 +198,30 @@ const Header = React.memo(
 							/>
 						)}
-						<HeaderTitle title={title} tmid={tmid} prid={prid} scale={scale} testID={testID} />
+						
```
Let's think about the HeaderTitle and how we can improve this code:

- We have a HeaderTitle component that can receive a component as the title;
- Now, we also have an onPress prop in this HeaderTitle component;

The best approach is to create a single component to pass as the `headerTitle` prop, which includes the `icon`, `title`, and `subtitle`, and to pass the `onPress` as a prop of the `HeaderTitle` component instead of creating a new `onTitlePress`.

OtavioStasiak commented on app/containers/RoomHeader/RoomHeader.tsx:
```diff
@@ -142,6 +117,7 @@ const Header = React.memo(
 		prid,
 		tmid,
 		onPress,
+		onTitlePress, 
```
Remove it.

OtavioStasiak commented on app/containers/Header/components/HeaderTitle/styles.ts:
```diff
@@ -6,20 +6,23 @@ export const styles = StyleSheet.create({
 		flex: 1,
 		alignItems: 'center',
-		justifyContent: 'center'
+		justifyContent: 'center',
```
Please, pay attention to the style changes.
The layout should not be altered.

| Develop | Your branch |
| ----- | ----- |
| <img width="1179" height="2556" alt="Simulator Screenshot - iPhone 16 - 2025-07-28 at 14 57 03" src="https://github.com/user-attachments/assets/fbfd307c-0612-4d8e-baa7-ef4b9b22cf89" /> | <img width="1179" height="2556" alt="Simulator Screenshot - iPhone 16 - 2025-07-28 at 14 57 09" src="https://github.com/user-attachments/assets/27be0eec-27d1-4e3e-ae69-908cb708fd59" /> |

OtavioStasiak commented on app/lib/PlatformPressable.tsx:
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
```
You should follow the current pattern used by the HeaderButtons. They use BorderlessButton, so you should use BorderlessButton as well instead of this archive.

abhishek0450 replied:
> You should follow the current pattern used by the HeaderButtons. They use BorderlessButton, so you should use BorderlessButton as well instead of this archive.

Thank you again for the thorough review, I really appreciate your patience.
I realize I introduced some avoidable mistakes while trying to fix earlier changes.
I’ll review everything more carefully now and update the PR shortly.
Thanks for helping me learn through this.
